### PR TITLE
New dump option to exclude entries whose value is the same as the default

### LINKED ISSUE #91: Support `--print_config skip_default`
Add support for `--print_config skip_default` so only the values different to the default are printed.

Two main benefits:
- Removes the manual work of removing the extra arguments from a config. However, this isn't recommended for reproducibility as it's useful to know exactly what was set.
- Improves backward compatibility. Often features are deprecated in a way that the default argument for a function is re-routed to the new implementation. Since the config saves all old defaults and passes them to the program, it is likely the program will fail when support for the old value is removed.

### PR DESCRIPTION
**Pull request by mauvilsa**
Fixes #91.

### CODE REVIEW

**Comment by mauvilsa** [comment]
@carmocca please review.

**Review by carmocca — APPROVED** [review]
Awesome!

Did not review the code but tested it with Lightning and it is working.

File: jsonargparse/actions.py [hunk 1]
```diff
@@ -244,12 +244,12 @@ def __init__(self,
...
                          default=default,
                          nargs=1,
-                         metavar='[={comments,skip_null}+]',
+                         metavar='[={comments,skip_null,skip_default}+]',
                          help='Print configuration and exit.')
 
     def __call__(self, parser, namespace, value, option_string=None):
         kwargs = {'subparser': parser, 'key': None, 'skip_none': False, 'skip_check': True}
-        valid_flags = {'': None, 'comments': 'yaml_comments', 'skip_null': 'skip_none'}
+        valid_flags = {'': None, 'comments': 'yaml_comments', 'skip_default': 'skip_default', 'skip_null': 'skip_none'}
```
[the remark this answers is not in the document]
Comment (mauvilsa) [reply jsonargparse/actions.py:252]: There is no way. And I think it does not make sense to have `skip_default` as default. If that were the default, then for any `CLI` if you do `--print_config` without any other arguments you would get an empty config. The main purpose of `--print_config` is to get a reference config for the entire parser. Including only what has changed is a different use case which is also good to have, but not the main one.

**Comment by mauvilsa** [comment]
> Awesome!
> 
> Did not review the code but tested it with Lightning and it is working.

I am fine with merging this now and include it in the next release. I think it is rather likely that it does not work as expected is complex cases, e.g. with many nested subclasses. That is why I asked you to test in a real use case. But we can fix issues as they are encountered.

**Pull request merged**